### PR TITLE
Replace `ui-constants` module with sidebar typedefs

### LINKED ISSUE #2927: Remove the `ui-constants` module
Maintaining, mocking and using the `ui-constants` module has been chafing us for a while. It could be replaced by a couple of `enum` types...

### PR DESCRIPTION
**Pull request by lyzadanger**
Pretty much like it says. Introduced a typedef "module" for the sidebar.

Fixes #2927

### CODE REVIEW

**Review by robertknight — COMMENTED** [review]
I had a few minor queries but overall this looks good.

File: src/types/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Type definitions for the sidebar
+ */
+
+/**
+ * Defined panel components available in the sidebar.
+ *
+ * @typedef { 'help' | 'loginPrompt' | 'shareGroupAnnotations' } PanelName
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/types/sidebar.js:8]: That's fine; I'll update. I wonder if there's any way _heck_ we could lint for this. Seems like something I'll probably repeat-offend on as I prefer spaciness in code, typically.

File: src/types/sidebar.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+/**
+ * @typedef { 'annotation' | 'note' | 'orphan' } TabName
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/types/sidebar.js:12]: Good catch! I'll track down and obliterate.

[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/types/sidebar.js:12]: Those were the original string values of the constants. We could pluralize if you want, but right now, I admit it sounds like an unfun busywork task :)

**Pull request merged**